backend/routes.py

```
@@ -828,9 +1255,279 @@ def run_manual_cleanup(dry_run: bool = True):
+    try:
+        cache = get_content_cache()
+        stats = cache.get_cache_stats()
+        return {"success": True, "stats": stats}
+    except Exception as e:
+        raise HTTPException(
+            status_code=500, detail=f"Failed to get cache stats: {str(e)}"
+        ) from e
+
+
+@router.post("/cache/cleanup")
+def cleanup_cache(data: dict = {}, authenticated: bool = Depends(verify_api_key)):
```

Ruff (0.12.2)

Do not use mutable data structures for argument defaults

Replace with `None`; initialize within function

(B006)